activestorage/app/models/active_storage/blob.rb

```diff
@@ -293,8 +293,9 @@ def delete
   # blobs. Note, though, that deleting the file off the service will initiate an HTTP connection to the service, which may
   # be slow or prevented, so you should not use this method inside a transaction or in callbacks. Use #purge_later instead.
   def purge
+    previously_persisted = persisted?
     destroy
-    delete
+    delete if previously_persisted
```

Comment:
I think this could be an Active Record method - we have `previously_new_record?` for example. However let's do that in a different PR.

Reply:
Good point, I'll make a follow up PR.